src/special_arrays.jl

```diff
@@ -111,6 +112,95 @@ function inverse_at!(x::AbstractVector, index, t::UnitVector, y::AbstractVector)
     index
 end
 
+####
+#### unit_vector_norm
+####
+
+struct UnitVectorNorm <: VectorTransform
+    n::Int
+    chi_prior::Bool
+    function UnitVectorNorm(n::Int; chi_prior::Bool = true)
+        @argcheck n ≥ 2 "Dimension should be at least 2."
+        new(n, chi_prior)
+    end
+end
+
+"""
+$(SIGNATURES)
+
+Transform `n ≥ 2` real numbers to a unit vector of length `n` and a radius, under the
+Euclidean norm. Returns the tuple `(normalized_vector, radius)`.
+
+When `chi_prior = true`, a prior correction is applied to the radius, which only
+affects the log Jacobian determinant. The purpose of this is to make the
+distribution proper. If you wish to use another prior, set this to `false` and use
+a manual correction, see also [`logprior`](@ref).
+
+!!! note
+    At the origin, this transform is non-bijective and non-differentiable. If
+    maximizing a target distribution whose density is constant for the unit vector,
+    then the maximizer using the Chi prior is at the origin, and behavior is undefined.
+
+!!! note
+    While ``n = 1`` would be technically possible, for practical purposes it would
+    likely suffer from numerical issues, since the transform is undefined at ``x = 0``,
+    and for a Markov chain to travel from ``y=[-1]`` to ``y=[1]``, it would have to leap
+    over the origin, which is only even possible due to discretization and likely will
+    often not work. Because of this, it is disallowed.
+"""
+unit_vector_norm(n::Int; chi_prior::Bool = true) = UnitVectorNorm(n; chi_prior)
+
+nonzero_logprior(t::UnitVectorNorm) = t.chi_prior
+
+function logprior(t::UnitVectorNorm, (y, r))
+    (; n, chi_prior) = t
+    if chi_prior
+        (t.n - 1) * log(r) - r^2 / 2
+    else
+        zero(r)
+    end
+end
+
+dimension(t::UnitVectorNorm) = t.n
+
+function _summary_rows(t::UnitVectorNorm, mime)
+    _summary_row(t, "$(t.n) element (unit vector, norm) transformation")
+end
+
+function transform_with(flag::LogJacFlag, t::UnitVectorNorm, x::AbstractVector, index)
+    (; n, chi_prior) = t
+    T = robust_eltype(x)
+    log_r = zero(T)
+    y = Vector{T}(undef, n)
+    copyto!(y, 1, x, index, n)
+    r = norm(y, 2)
+    __normalize!(y, r)
+    ℓ = flag isa NoLogJac ? flag : (chi_prior ? -r^2 / 2 : -(t.n - 1) * log(r))
+    index += n
+    (y, r), ℓ, index
+end
+
+function inverse_eltype(t::UnitVectorNorm,
+                        ::Type{Tuple{V,T}}) where {V <: AbstractVector,T}
+    _ensure_float(eltype(T))
+end
+
+function inverse_at!(x::AbstractVector, index, t::UnitVectorNorm,
+                     (y, r)::Tuple{AbstractVector,Real})
+    (; n) = t
+    @argcheck length(y) == n
+    @argcheck r ≥ 0
+    _x = @view x[index:(index + n - 1)]
+    if r == 0
+        _x .= zero(eltype(x))
```

Comment:
Maybe use `fill!` to avoid broadcasting?
```suggestion
        fill!(_x, zero(eltype(x))
```